pkg/auth/auth.go

```diff
@@ -10,95 +10,105 @@ import (
 )
 
 const (
-	activeDirectoryEndpoint                         = "https://login.microsoftonline.com/"
-	adalTokenFromMSIOperationName                   = "adal_token_msi"
-	adalTokenFromMSIWithUserAssignedIDOperationName = "adal_token_msi_userassignedid"
-	adalTokenOperationName                          = "adal_token"
+	activeDirectoryEndpoint = "https://login.microsoftonline.com/"
 )
 
 var reporter *metrics.Reporter
 
 // GetServicePrincipalTokenFromMSI return the token for the assigned user
 func GetServicePrincipalTokenFromMSI(resource string) (*adal.Token, error) {
 	begin := time.Now()
+	var err error
+
+	defer func() {
+		if err != nil {
+			reporter.ReportIMDSOperationError(metrics.AdalTokenFromMSIOperationName)
+			return
+		}
```

Comment:
Shouldn't we count the operations themselves or is the idea to look at the latency reports and find the total number based on the key value of count in the latency metrics.